src/main/java/com/lambdatest/jenkins/freestyle/report/AppAutomationReportBuildAction.java

```diff
@@ -0,0 +1,135 @@
+                        char[] charArrayForSession = new char[1024];
+                        sb = new StringBuffer();
+                        while ((numCharsReadForSession = isr.read(charArrayForSession)) > 0) {
+                            sb.append(charArrayForSession, 0, numCharsReadForSession);
+                        }
+                        String sessionResult = sb.toString();
+
+                        JsonNode sessionRootNode = mapper.readTree(sessionResult);
+                        ArrayNode sessionDataNode = (ArrayNode) sessionRootNode.get("data");
+                        Iterator<JsonNode> sessionDataIterator = sessionDataNode.elements();
+
+                        JSONArray array = new JSONArray();
+                        while (sessionDataIterator.hasNext()) {
+                            JsonNode sessionDetailNode = sessionDataIterator.next();
+                            logger.info("sessiondetailnode : " + sessionDataNode.toString());
+                            String testUrl = "https://appautomation.lambdatest.com/test?testID=" + sessionDetailNode.get("test_id").toString().replaceAll("\"", "") + "&buildid=" + build_id;
+                            logger.info("testUrl : " + testUrl);
+                            JSONObject resultJSON = new JSONObject();
+                            resultJSON.put("url",testUrl);
+                            resultJSON.put("status",sessionDetailNode.get("status_ind").toString().replaceAll("\"", ""));
+                            resultJSON.put("deviceVersion",sessionDetailNode.get("version").toString().replaceAll("\"", ""));
+                            resultJSON.put("OS",sessionDetailNode.get("platform").toString().replaceAll("\"", ""));
+                            resultJSON.put("name",sessionDetailNode.get("name").toString().replaceAll("\"", ""));
+                            resultJSON.put("testDuration",sessionDetailNode.get("duration").toString().replaceAll("\"", ""));
+                            resultJSON.put("createdAtReadable",sessionDetailNode.get("create_timestamp").toString().replaceAll("\"", ""));
+                            array.put(resultJSON);
+                        }
+                        for (int i = 0; i < array.length(); i++) {
+                            result.add(array.getJSONObject(i));
+                        }
+                        logger.info("result : " + result.toString());
+                        break;
+                    }
+                }
+            } catch (JsonParseException e) {
+                e.printStackTrace();
+            } catch (JsonMappingException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Please add `logger.error` statements for  `e.printStackTrace()`

Reply:
Done